web/blog/2025-06-20-oauth-security-incident.md

```diff
@@ -0,0 +1,101 @@
+---
+title: 'Incident Report: Case insensitive OAuth IDs  vulnerability in Wasp'
+authors: [miho]
+image: /img/security-incident/security-incident.png
+tags: [security, oauth, keycloak, wasp]
+---
+
+## Introduction
+
+We received a report on a security vulnerability in Wasp auth related to our OAuth support in Wasp `0.16.5` and earlier.
+
+**This issue only affects users using Keycloak authentication with non-default case-sensitive IDs.**
+
+All other configurations (Google, GitHub, Discord, or Keycloak with the default case-insensitive configuration) are **not affected**. Email and username auth providers are also **not affected**.
+
+Users should upgrade to Wasp `0.16.6` which contains the fix ASAP.
```

Comment:
They will also need to migrate their data in the database, correct? Since provider ids were saved lowercased so far? Or hm they can't even migrate it, because there is information loss -> they can't "unlowercase it". What should they do then? What is with the currently saved accounts in the database, that are using Keycloak with case sensitive identifiers, what should they doa bout them? I would hint about it here, and give some more pointers down in the solution.